x-pack/plugins/cases/server/features/v2.ts

```diff
@@ -0,0 +1,195 @@
+      insightsAndAlerting: [APP_ID],
+    },
+    cases: [APP_ID],
+    privileges: {
+      all: {
+        api: apiTags.all,
+        cases: {
+          create: [APP_ID],
+          read: [APP_ID],
+          update: [APP_ID],
+          push: [APP_ID],
+        },
+        management: {
+          insightsAndAlerting: [APP_ID],
+        },
+        savedObject: {
+          all: [...filesSavedObjectTypes],
+          read: [...filesSavedObjectTypes],
+        },
+        ui: capabilities.all,
+      },
+      read: {
+        api: apiTags.read,
+        cases: {
+          read: [APP_ID],
+        },
+        management: {
+          insightsAndAlerting: [APP_ID],
+        },
+        savedObject: {
+          all: [],
+          read: [...filesSavedObjectTypes],
+        },
+        ui: capabilities.read,
+      },
+    },
+    subFeatures: [
+      {
+        name: i18n.translate('xpack.cases.features.deleteSubFeatureName', {
+          defaultMessage: 'Delete',
+        }),
+        privilegeGroups: [
+          {
+            groupType: 'independent',
+            privileges: [
+              {
+                api: apiTags.delete,
+                id: CASES_DELETE_SUB_PRIVILEGE_ID,
+                name: i18n.translate('xpack.cases.features.deleteSubFeatureDetails', {
+                  defaultMessage: 'Delete cases and comments',
+                }),
+                includeIn: 'all',
+                savedObject: {
+                  all: [...filesSavedObjectTypes],
+                  read: [...filesSavedObjectTypes],
+                },
+                cases: {
+                  delete: [APP_ID],
+                },
+                ui: capabilities.delete,
+              },
+            ],
+          },
+        ],
+      },
+      {
+        name: i18n.translate('xpack.cases.features.casesSettingsSubFeatureName', {
+          defaultMessage: 'Case settings',
+        }),
+        privilegeGroups: [
+          {
+            groupType: 'independent',
+            privileges: [
+              {
+                id: CASES_SETTINGS_SUB_PRIVILEGE_ID,
+                name: i18n.translate('xpack.cases.features.casesSettingsSubFeatureDetails', {
+                  defaultMessage: 'Edit case settings',
+                }),
+                includeIn: 'all',
+                savedObject: {
+                  all: [...filesSavedObjectTypes],
+                  read: [...filesSavedObjectTypes],
+                },
+                cases: {
+                  settings: [APP_ID],
+                },
+                ui: capabilities.settings,
+              },
+            ],
+          },
+        ],
+      },
+      {
+        name: i18n.translate('xpack.cases.features.addCommentsSubFeatureName', {
+          defaultMessage: 'Create comments & attachments',
+        }),
+        privilegeGroups: [
+          {
+            groupType: 'independent',
+            privileges: [
+              {
+                api: apiTags.all,
```

Comment:
Wouldn't that require: adding tags to all comments routes and making use of them, and also updating all the documentation to make that clear, as both here https://www.elastic.co/guide/en/kibana/current/manage-cases.html and in the code here https://github.com/elastic/kibana/blob/main/x-pack/plugins/cases/server/routes/api/comments/post_comment.ts#L28 we specifically state 'you must have all'? Seems sort of out of scope of this change, as tags are not used in any other comments routes/sparingly in cases routes at all, and we don't gain much besides a 403 unauthorized instead of whatever ensureAuthorized throws

Reply:
Here we configure the feature privilege to tell for which APIs the user has access to. One of the API tags that are exported by `apiTags.all` is the one related to uploading files (not the Cases API but the one exported by the File service). Now that we have introduced the new subfeature privilege to prevent users from adding comments and attachments with the "All" privilege we should also split the tags so users with "All" cannot use the File service APIs to upload files specific to cases. If we do not split the tags, a user with "All" but not "Create comment" could upload a file which is something we do not want.


Before the introduction of the new subfeature privilege

Reply:
Tested this via api only, and we do indeed block it

Reply:
```
kibana git:(cases-subfeatures-main) curl 'http://localhost:5601/internal/cases/c24034da-ab81-4000-ae69-813eefb4da8a/attachments/_bulk_create' \
  -H 'Accept: */*' \
  -H 'Accept-Language: en-US,en;q=0.9' \                                                                                  -H 'Cache-Control: no-cache' \                                                                                          -H 'Connection: keep-alive' \                                                                                           -H 'Content-Type: application/json' \                                                                                                    -H 'Origin: http://localhost:5601' \
  -H 'Pragma: no-cache' \
  -H 'Referer: http://localhost:5601/app/management/insightsAndAlerting/cases/c24034da-ab81-4000-ae69-813eefb4da8a/?tabId=files' \                                                                                                                -H 'Sec-Fetch-Dest: empty' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Site: same-origin' \
  -H 'Sec-GPC: 1' \                                                                                                       -H 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/130.0.0.0 Safari/537.36' \                                                                                                              -H 'kbn-build-number: 9007199254740991' \                                                                               -H 'kbn-version: 9.0.0' \                                                                                               -H 'sec-ch-ua: "Chromium";v="130", "Brave";v="130", "Not?A_Brand";v="99"' \
  -H 'sec-ch-ua-mobile: ?0' \                                                                                             -H 'sec-ch-ua-platform: "Windows"' \                                                                                    -H 'x-elastic-internal-origin: Kibana' \                                                                                -H 'x-kbn-context: %7B%22type%22%3A%22application%22%2C%22name%22%3A%22management%22%2C%22url%22%3A%22%2Fapp%2Fmanagement%2Fsecurity%2Froles%22%7D' \                                                                                   <....
{"statusCode":403,"error":"Forbidden","message":"Unauthorized to create comments with owners: \"cases\""}%
```

Reply:
The Kibana File Service exposes some APIs to upload files. The APIs are of the form `api/files/files/casesFilesCases` and they are bound to Cases. These APIs are unrelated to creating comments or bulk creating attachment APIs. We use the `access:` API tags to give access to the File APIs to users when they have access to cases. Otherwise, they cannot upload a file. Now that we split up the features and we have a dedicated subfeature privilege for creating a comment (synonymous to uploading a file) we should also split up the tags and the `access:` tag for using the file API should be part of the new subfeature privilege and not the "All" privilege. The reason is that if I do not have the "Create comment & attachment" subfeature privilege I should not be able to use the `api/files/files/casesFilesCases` to upload the file.

About blocking the `bulkCreateComment` this is fine. If users do not have the "Create comment & attachment" subfeature privilege they should not be able to use the API and they should get a 403.

Reply:
done c5688d517300888c6127014a56d0d791d1167540 I mistakenly assumed file permissions were controlled by 
![image](https://github.com/user-attachments/assets/3f7e4d0d-ab3f-483a-b663-5a84469b0d6d)
and did not see that the api property of the privilege group was automagically injected into the case's own file routes, and had nothing to do with any of the case route tags as of now.